enterprise/frontend/src/metabase-enterprise/collections/utils.ts

```diff
@@ -72,3 +75,26 @@ export const getIcon = (item: ObjectWithModel): IconData => {
 
   return getIconBase(item);
 };
+
+/** Removes items from the array that belong to the instance analytics collection or one of its children */
+export const filterOutItemsFromInstanceAnalytics = <
+  Item extends ItemWithCollection,
+>(
+  items: Item[],
+) => {
+  /** Cache of ids of instance analytics collections */
+  const cache = new Set<CollectionId>();
+
+  return items.filter(item => {
+    if (cache.has(item.collection.id)) {
+      return false;
+    }
+    const ancestors = item.collection.effective_ancestors || [];
+    const path = [item.collection, ...ancestors];
+    if (path.some(isInstanceAnalyticsCollection)) {
+      path.map(c => c.id).forEach(id => cache.add(id));
+      return false;
+    }
+    return true;
+  });
+};
```

Comment:
Since effective ancestors starts from the root, we could just check without having to build up a cache like this

```
export const filterOutItemsFromInstanceAnalytics = <
  Item extends ItemWithCollection,
>(
  items: Item[],
) => {
  const collection = item.collection;
  const firstAncestor = _.first(item.collection.effective_ancestors);

  return items.filter(item => isInstanceAnalyticsCollection(collection) || (firstAncestor && isInstanceAnalyticsCollection(firstAncestor))
});
```

I think this code is simpler for the next person (not as worried about perf, you've solve that with a cache here).